### PR TITLE
ORC-737: Upgrade Spark to 3.1.0

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade Spark from 2.4.6 to 3.1.0 in benchmark module.
Please note that we will upgrade to Spark 3.1.1 immediately as soon as possible because 3.1.0 is not an official release.

### Why are the changes needed?

To use the latest version in benchmark.

### How was this patch tested?

```bash
$ cd java/bench

$ mvn package
INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for ORC Benchmarks 1.7.0-SNAPSHOT:
[INFO] 
[INFO] ORC Benchmarks ..................................... SUCCESS [  0.891 s]
[INFO] ORC Benchmarks Core ................................ SUCCESS [  5.977 s]
[INFO] ORC Benchmarks Hive ................................ SUCCESS [  6.855 s]
[INFO] ORC Benchmarks Spark ............................... SUCCESS [ 26.922 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  40.809 s
[INFO] Finished at: 2021-01-17T20:28:48-08:00
[INFO] ------------------------------------------------------------------------

$ java -jar spark/target/orc-benchmarks-spark-*.jar spark -i 1 -I 1 ~/data    
# JMH version: 1.20
# VM version: JDK 1.8.0_275, VM 25.275-b01
# VM invoker: /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
# VM options: -server -Xms256m -Xmx2g -Dbench.root.dir=/home/dongjoon/data
# Warmup: 1 iterations, 10 s each
# Measurement: 1 iterations, 10 s each
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Average time, time/op
# Benchmark: org.apache.orc.bench.spark.SparkBenchmark.fullRead
# Parameters: (compression = none, dataset = taxi, format = orc)

# Run progress: 0.00% complete, ETA 00:27:00
# Fork: 1 of 1
# Warmup Iteration   1: [WARN ] Unable to load native-hadoop library for your platform... using builtin-java classes where applicable

Records: 22773249
Invocations: 1
Reads: 81
Bytes: 1333068794
11158424.688 us/op
Iteration   1: 
Records: 45546498
Invocations: 2
Reads: 162
Bytes: 2666137588
9776462.312 us/op
                 bytesPerRecord: 58.537 #
                 perRecord:      0.429 us/op
                 reads:          81.000 #
                 records:        22773249.000 #
...
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @pgaref

**Comment by pgaref** [comment]
Thanks @dongjoon-hyun  -- we should definitely bump Spark to 3 but as you mentioned it needs to be 3.1.1 as 3.1.0 had some unresolved issues straight from release if I recall correctly from the email thread.

So +1 from my side for 3.1.1 when it gets published

**Comment by dongjoon-hyun** [comment]
Thank you, @pgaref . Yes, this PR will wait for the official 3.1.1 release.

File: java/bench/spark/pom.xml [hunk 1]
```diff
@@ -135,6 +134,21 @@
       <groupId>org.scala-lang</groupId>
       <artifactId>scala-library</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-core</artifactId>
+      <version>2.10.0</version>
```
Comment (omalley) [inline java/bench/spark/pom.xml:140]: Can you please define a variable to hold the jackson version?

Comment (dongjoon-hyun) [reply java/bench/spark/pom.xml:140]: Sure!

**Comment by dongjoon-hyun** [comment]
Thank you, @omalley !

**Pull request closed without merging**